src/SDiagonal.jl

```diff
@@ -17,6 +17,10 @@ SDiagonal(a::StaticMatrix{N,N,T}) where {N,T} = Diagonal(diag(a))
 size(::Type{<:SDiagonal{N}}) where {N} = (N,N)
 size(::Type{<:SDiagonal{N}}, d::Int) where {N} = d > 2 ? 1 : N
 
+Base.axes(D::SDiagonal, d) = d <= 2 ? axes(D)[d] : SOneTo(1)
+
+Base.reshape(a::SDiagonal, s::Tuple{SOneTo,Vararg{SOneTo}}) = reshape(a, homogenize_shape(s))
```

Comment:
This is the same as the generic fallback:
https://github.com/JuliaArrays/StaticArrays.jl/blob/29a76ec7fdd675c9982495f9898f5ada77476b9d/src/abstractarray.jl#L190 .
Do we really need this method?

Reply:
Other than this small point, the PR looks nice :+1: .